### PR TITLE
Add support for using reverse scans to satisfy ORDER BY.

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

File: sql/values.go [hunk 1]
```diff
@@ -71,3 +72,38 @@ func (n *valuesNode) Next() bool {
 func (*valuesNode) Err() error {
 	return nil
 }
+
+func (n *valuesNode) Len() int {
```
Comment (petermattis) [inline sql/values.go]: This was all just code movement from `sortNode`.

**Comment by tamird** [comment]
LGTM

File: sql/select.go [hunk 2]
```diff
@@ -423,18 +425,20 @@ func (m qvalueRangeMap) getRange(colID ColumnID) *qvalueRange {
...
-	index *IndexDescriptor
-	start []qvalueInfo
-	end   []qvalueInfo
-	cost  float64
+	desc    *TableDescriptor
+	index   *IndexDescriptor
+	start   []qvalueInfo
+	end     []qvalueInfo
+	cost    float64
+	viable  bool
```
Comment (tschottdorf) [inline sql/select.go]: could you add a comment for `viable`?

Comment (petermattis) [inline sql/select.go]: Done. And renamed to `covering`.

**Pull request merged**